pkg/operator/k8sutil/job.go

```diff
@@ -103,8 +103,8 @@ func DeleteBatchJob(clientset kubernetes.Interface, namespace, name string, wait
 		return nil
 	}
 
-	retries := 20
-	sleepInterval := 2 * time.Second
+	retries := 30
+	sleepInterval := 3 * time.Second
```

Comment:
The only thing I could suggest is adding a comment here that explains this case so we don't change this in the future. 

I'm not sure if it would be truly useful to add unit/integration tests for this given that it is really about the behavior of kubernetes and the code's understanding of it.